### ISSUE TITLE
Adding Matter Devices failed because tried to connect to an old not existing Network

### ISSUE THREAD

**Issue by vega82** [home-assistant/addons#3823]
### Describe the issue you are experiencing

After deleting the (by Matter installation/configuration) preconfigured Thread Network and recreate a new network with a different Name, Matter devices allready try to connect to the old Network Name during the adding process.

"No connection to Thread-Network "ha-thread-xxxx" possible
While xxxx is the old network Name.

I currently use HA 2024.10.4
Problem allready exists in releases before.

### What type of installation are you running?

Home Assistant Supervised

### Which operating system are you running on?

Home Assistant Operating System

### Which add-on are you reporting an issue with?

OpenThread Border Router

### What is the version of the add-on?

2.11.1

### Steps to reproduce the issue

1. Add a new Matter device
2. HA tries to find Device
3. Matter exchanges the Connection Data
4. Process failes with "No connection to Thread-Network "ha-thread-xxxx" possible

### System Health information

healthy

**Comment by Neon-44** [comment]
exactly the same problem. Did you ever fix it?

**Comment by MackenzieMolloy** [comment]
Hello, I had the same issue and after a lot of debugging managed to fix this.

For context, I previously had a Thread Network running along with some Matter-enabled devices connected... and then went to re-setup everything as I was changing the Raspberry PI everything was running on among other things.

To fix the issue, I did the following:
- **Cleared Google Play Services Data & Cache**: On my Andriod device, which was used to Add Matter Devices, I cleared the `Google Play Services` Data and Cache via the buttons in `Settings > Apps > Google Play Services > Storage`.
- **Re-installed Google Home**: For some reason, Matter support was unavailable after clearing the Data & Cache above and re-installing `Google Home` re-enabled Matter support.
- **Re-installed HomeAssistant App**: Not sure if I needed to but I was trying everything possible at the time to get things working.
- **Sync Thread credentials**: Synced Thread Credentials via the Home Assistant App on my Andriod device, which was used to Add Matter Devices, via ``Settings > Companion App > Troubleshooting > Sync Thread credentials`.

After doing all the above, I restarted my Phone and Raspberry PI running home-assistant and then was able to successfully add Matter-enabled devices to my active thread network. :)
